impeller/entity/contents/filters/blend_filter_contents.cc

```diff
@@ -178,6 +179,168 @@ static std::optional<Entity> AdvancedBlend(
+        break;
+      case BlendMode::kSoftLight:
+        cmd.pipeline = renderer.GetBlendSoftLightPipeline(options);
+        break;
+      case BlendMode::kDifference:
+        cmd.pipeline = renderer.GetBlendDifferencePipeline(options);
+        break;
+      case BlendMode::kExclusion:
+        cmd.pipeline = renderer.GetBlendExclusionPipeline(options);
+        break;
+      case BlendMode::kMultiply:
+        cmd.pipeline = renderer.GetBlendMultiplyPipeline(options);
+        break;
+      case BlendMode::kHue:
+        cmd.pipeline = renderer.GetBlendHuePipeline(options);
+        break;
+      case BlendMode::kSaturation:
+        cmd.pipeline = renderer.GetBlendSaturationPipeline(options);
+        break;
+      case BlendMode::kColor:
+        cmd.pipeline = renderer.GetBlendColorPipeline(options);
+        break;
+      case BlendMode::kLuminosity:
+        cmd.pipeline = renderer.GetBlendLuminosityPipeline(options);
+        break;
+      default:
+        return false;
+    }
+
+    FS::BlendInfo blend_info;
+    VS::FrameInfo frame_info;
+
+    auto dst_sampler_descriptor = dst_snapshot->sampler_descriptor;
+    if (renderer.GetDeviceCapabilities().SupportsDecalTileMode()) {
+      dst_sampler_descriptor.width_address_mode = SamplerAddressMode::kDecal;
+      dst_sampler_descriptor.height_address_mode = SamplerAddressMode::kDecal;
+    }
+    auto dst_sampler = renderer.GetContext()->GetSamplerLibrary()->GetSampler(
+        dst_sampler_descriptor);
+    FS::BindTextureSamplerDst(cmd, dst_snapshot->texture, dst_sampler);
+    frame_info.dst_y_coord_scale = dst_snapshot->texture->GetYCoordScale();
+    blend_info.dst_input_alpha =
+        absorb_opacity ? dst_snapshot->opacity * alpha.value_or(1.0) : 1.0;
+
+    blend_info.color_factor = 1;
+    blend_info.color = foreground_color;
+    // This texture will not be sampled from due to the color factor. But
+    // this is present so that validation doesn't trip on a missing
+    // binding.
+    FS::BindTextureSamplerSrc(cmd, dst_snapshot->texture, dst_sampler);
+
+    auto blend_uniform = host_buffer.EmplaceUniform(blend_info);
+    FS::BindBlendInfo(cmd, blend_uniform);
+
+    frame_info.mvp = Matrix::MakeOrthographic(pass.GetRenderTargetSize());
+
+    auto uniform_view = host_buffer.EmplaceUniform(frame_info);
+    VS::BindFrameInfo(cmd, uniform_view);
+
+    return pass.AddCommand(cmd);
+  };
+  CoverageProc coverage_proc =
+      [coverage](const Entity& entity) -> std::optional<Rect> {
+    return coverage;
+  };
+
+  auto contents = AnonymousContents::Make(render_proc, coverage_proc);
+
+  // If there is pending opacity but it was not absorbed by this entity, we have
+  // to convert this back to a snapshot so it can be passed on. This generally
+  // implies that there is another filter about to run, so we'd perform this
+  // operation anyway.
+  auto potential_opacity = alpha.value_or(1.0) * dst_snapshot->opacity;
```

Comment:
This seems like a reasonable thing to do?

Reply:
Yeah I think this makes sense, although it'll probably just end up getting absorbed  right away due to no filter being chained most of the time. Maybe we can come up with a mechanism to reconcile this later.